### ISSUE TITLE
[Bug]: Polar plot error bars don't rotate with angle for `set_theta_direction` and `set_theta_offset`

### ISSUE THREAD

**Issue by th0ger** [matplotlib/matplotlib#27784]
### Bug summary

Reopening #441 since the fix seems not to work on two corner cases:
- Clockwise direction with `set_theta_direction(-1)`
- Angle offset set by `set_theta_offset(np.pi/2)`

### Code for reproduction

```python
from numpy import *
from matplotlib import pyplot as py

r = zeros([16])
err = zeros([16])
for i in range(16):
    r[i] = random.randint(400, 600)
    err[i] = random.randint(20, 50)
theta = arange(0, 2 * pi, 2 * pi / 16)
width = pi * 2 / 16

fig = py.figure(figsize=(8, 8))
ax = fig.add_axes([0.1, 0.1, 0.75, 0.79], polar=True)

bars = ax.bar(
    theta + pi / 16,
    r,
    width=width,
    bottom=0.0,
    yerr=err,
    capsize=15 # Capsize set to exaggerate issue
    )  
ax.set_ylim(0, 700)

ax.set_theta_direction(-1)  # Clockwise direction

py.show()
```

```python
from numpy import *
from matplotlib import pyplot as py

r = zeros([16])
err = zeros([16])
for i in range(16):
    r[i] = random.randint(400, 600)
    err[i] = random.randint(20, 50)
theta = arange(0, 2 * pi, 2 * pi / 16)
width = pi * 2 / 16

fig = py.figure(figsize=(8, 8))
ax = fig.add_axes([0.1, 0.1, 0.75, 0.79], polar=True)

bars = ax.bar(
    theta + pi / 16,
    r,
    width=width,
    bottom=0.0,
    yerr=err,
    capsize=15 # Capsize set to exaggerate issue
    )  
ax.set_ylim(0, 700)

ax.set_theta_offset(np.pi/2)  # Start from the top

py.show()
```

### Actual outcome

![image](https://github.com/matplotlib/matplotlib/assets/26322144/2f470802-4e8d-4a3b-af7a-da3eed58845e)
![image](https://github.com/matplotlib/matplotlib/assets/26322144/a40a555b-17e1-43ae-a080-8396a2186068)

### Expected outcome

Errorbar caps perpendicular to the error bars.

### Operating system

Ubuntu

### Matplotlib Version

3.8.1

### Matplotlib Backend

agg

### Python version

3.11.4

### Jupyter version

7.0.6

### Installation

None

**Comment by dstansby** [comment]
I just had a go at this, and it's not a simple fix because the errorbar rotations are done before drawing, but the Axes offset/direction are only known/set at draw time.

**Comment by dstansby** [comment]
Thanks for the clear report though!

**Comment by QuLogic** [comment]
This was fixed by #28966.
